### DISCUSSION TITLE
How to use it with an existing database?

### DISCUSSION THREAD

**Discussion by biapar** [pdevito3/craftsman#132] [Q&A] [answered]
How to use it with an existing database?

**Comment by pdevito3** [comment]
I need to make assumptions when you say database given the lack of detail in the question here. If you actually mean a database then you can use the connection string for the db like you normally would. Add a custom schema if you want to keep things isolated. 

If you mean an existing project, generally craftsman will be used most successfully with a scaffolded wrapt project. If you have some recognizable paths it might be able to scaffold out some stuff for you, but idk if that'll be worth all the reconciliation time. 

In the future, more detail than a single sentence would be helpful so I can give a better response.

**Reply by biapar** [reply]
Thank you for your reply. I mean if I've already a database with its schema and tables + data.

**Reply by pdevito3** [reply]
Yeah it's not really meant for that kind of use case.

Another option could be to make a new Wrapt project with craftsman and do a db first migration, move the entities to a distinct folder, incrementally move to code first entities from craftsman.

**Reply by biapar** [reply]
Maybe a feature to introduce. 
Did you tried on an existing db if EF made a reverse of the db?

**Reply by pdevito3** [reply, marked as answer]
Like I mentioned, depending on your situation, you should be able to do something along the lines of doing db first in an empty project. That will give you simple anemic entities and then you incrementally transition to craftsman scaffolded entities over time.

Practically you'll have a lot of business logic in an existing project though so pulling over the db repetition is just a small part of the picture and probably won't be enough. 

That's a big reason this isn't an existing feature. Every project is different and wouldn't have a seamless plug and play transition

**Reply by biapar** [reply]
No. Today I've an old WebApi business logic that "talks" with the db. I wish to migrate on a new architecture. I can to what you said and then "copy" data from one db to another or change the connection. I'll try and I'll update you which solution I'll use.
Thanks a lot also for the very smart tool!
